lib/cli/node-modules-path.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+var path  = require('path');
+
+function isSubdirectoryOf(parentPath, possibleChildPath) {
+  return possibleChildPath.length > parentPath.length &&
+    possibleChildPath.indexOf(parentPath) === 0;
+}
+
+// A utility function for determining what path an addon may be found at. Addons
+// will only be resolved in the project's local `node_modules/` directory. This
+// is in contrast to a plain `require('some-module')` lookup, which would resolve
+// a library according to the paths in NODE_PATH.
+//
+// A description of node's lookup logic can be found here:
+//
+// * https://nodejs.org/api/modules.html#modules_all_together
+//
+// By requiring with an absolute path this logic is bypassed.
+//
```

Comment:
can this be converted into the YUIdoc form, this will match other documented functions.

we are slowly making progress on improving doc coverage: https://inch-ci.org/github/ember-cli/ember-cli?pending_build=47203